### ISSUE TITLE
fieldnames can give incorrect generated types

### ISSUE THREAD

**Issue by maapteh** [dotansimha/graphql-code-generator#1365]
**Describe the bug**
fieldnames can give incorrect generated types.

**To Reproduce**
When i have a 'mutation' field in the type 'BasketMutationResult' i get the following result:

```
export namespace AddBasketItem {
    export type Variables = {
        item: BasketMutation;
    };

    export type Mutation = {
        __typename?: 'Mutation';

        addBasketItem: Maybe<AddBasketItem>;
    };

    export type AddBasketItem = {
        __typename?: 'BasketMutationResult';

        mutation: Maybe<Mutation>;
    };

    export type Mutation =
        | BasketMutationType.Fragment
        | BasketDeleteType.Fragment;
}
```
It adds type Mutation twice while the last is not correct.

**Expected behavior**
Not do this :)

**Environment:**
 - OS: OSX
 - Codegen: 0.17.0
 - Node: 10

**Comment by kamilkisiela** [comment]
#1369 should fix it

**Comment by maapteh** [comment]
super!

**Comment by dotansimha** [comment]
Fixed in 0.18.0 🎉
